tests/integration.rs

```diff
@@ -460,6 +464,77 @@ fn fw_path(_fw_type: FwType) -> String {
     fw_path.to_str().unwrap().to_string()
 }
 
+struct MetaEvent {
+    event: String,
+    device_id: Option<String>,
+}
+
+impl MetaEvent {
+    pub fn match_with_json_event(&self, v: &serde_json::Value) -> bool {
+        let mut matched = false;
+        if v["event"].as_str().unwrap() == self.event {
+            if let Some(device_id) = &self.device_id {
+                if v["properties"]["id"].as_str().unwrap() == device_id {
+                    matched = true
+                }
+            } else {
+                matched = true;
+            }
+        }
+        matched
+    }
+}
+
+// Parse the event_monitor file based on the format that each event
+// is surrounded by '{' and '}'
+fn parse_event_file(event_file: &str) -> Vec<serde_json::Value> {
```

Comment:
Can't you rely a bit more on `serde_json` to simplify the parsing?

Reply:
That's what I tried at the beginning and end up giving up. The problem was that the event-monitor output file is not valid JSON format file (instead it is a collection of valid JSON entries). Making it a valid JSON file throughout the VMM lifetime would need both changes from the `event_monitor` and pre-processing from the test (that is slightly simpler than the current parsing).

Reply:
Gotcha. Well it's up to you. Maybe prototype this as well in a separate PR.

Reply:
I think that if there were more newlines in the event output then the parsing would be easier.

Reply:
That's indeed a simpler solution. PR Updated. PTAL.